odl/tomo/operators/xray_trafo.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""X-ray transforms."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from future.builtins import str, super
+
+# External
+import numpy as np
+
+# Internal
+from odl.discr.lp_discr import DiscreteLp
+from odl.space.fspace import FunctionSpace
+from odl.operator.operator import Operator
+from odl.space.cu_ntuples import CUDA_AVAILABLE
+if CUDA_AVAILABLE:
+    from odl.space.cu_ntuples import CudaNtuples
+else:
+    CudaNtuples = type(None)
+from odl.tomo.geometry.geometry import Geometry
+from odl.tomo.backends.astra_cpu import ASTRA_AVAILABLE
+from odl.tomo.backends.astra_cuda import ASTRA_CUDA_AVAILABLE
+if ASTRA_AVAILABLE:
```

Comment:
Shouldnt need to fence these, they should exist (and throw if they do not work).

This is a more simple pattern to work with since we dont need to put fences everywhere.